src/server/handlers/launchpad.js

```diff
@@ -612,19 +665,37 @@ export const authorizeSnap = async (req, res) => {
   const macaroon = req.body.macaroon;
 
   try {
-    await checkAdminPermissions(req.session, repositoryUrl);
+    const { owner, name } = await checkAdminPermissions(
+      req.session, repositoryUrl
+    );
     const result = await internalFindSnap(repositoryUrl);
     // Registration happened a short while ago; we increment the metric here
     // in order to support having the client call `register-name` on the
     // store directly rather than going via our backend.  As such, we don't
     // need to roll this back if authorization fails.
     await db.transaction(async (trx) => {
+      let dbUser = null;
       if (req.session && req.session.user) {
-        await db.model('GitHubUser').incrementMetric(
+        dbUser = await db.model('GitHubUser').incrementMetric(
           { github_id: req.session.user.id }, 'names_registered', 1,
           { transacting: trx }
         );
       }
+      const repositoryModel = db.model('Repository');
```

Comment:
Similar example here. 

As long as the metrics related code was just single call of `await db.model('GitHubUser').incrementMetric(...)` it was pretty short and didn't confuse when reading `authorizeSnap` handler.
Now metrics part grows to ~20 lines of code making it harder to spot what part is really about authorizing snap and which is keeping db up to date.